pkg/util/kubernetes/node.go

```diff
@@ -0,0 +1,52 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"context"
+	"github.com/apache/camel-k/v2/pkg/client"
+	"github.com/apache/camel-k/v2/pkg/util/sets"
+	"github.com/apache/camel-k/v2/pkg/util/slice"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NodeListLookup lookup the list of nodes in the cluster
+func NodeListLookup(ctx context.Context, c client.Client) ([]corev1.Node, error) {
+	nodes, err := c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return nodes.Items, err
```

Comment:
nitpick: usually it is better to explicitly return `nil` if there are no error and the reason is that, [a nil error value not equal to nil]( https://go.dev/doc/faq#nil_error). In this snipped, it won't make a difference as `err` is never reused, but to avoid subtle bugs, I would recommend explicitly return `nil`

Reply:
good catch - thanks for the insight